src/method.rs

```diff
@@ -0,0 +1,309 @@
+                for (wszname, value) in in_params {
+                    let wszname = HSTRING::from(wszname);
+                    let value = TryInto::<VARIANT>::try_into(value.clone())?;
+
+                    // See https://learn.microsoft.com/en-us/windows/win32/api/wbemcli/nf-wbemcli-iwbemclassobject-put
+                    // Note that the example shows the variant is expected to be cleared (dropped) after the call to Put,
+                    // so passing &value is acceptable here
+                    unsafe {
+                        inst.Put(&wszname, Default::default(), &value, 0)?;
+                    }
+                }
+                Some(inst)
+            }
+            None => None,
+        };
+
+        // In the case of a method with no out parameters and a VOID return type, there will be no out-parameters object
+        let mut output = None;
+        unsafe {
+            self.svc.ExecMethod(
+                &object_path,
+                &method,
+                Default::default(),
+                &self.ctx.0,
+                in_params.as_ref(),
+                Some(&mut output),
+                None,
+            )?;
+        }
+
+        Ok(output.map(IWbemClassWrapper::new))
+    }
+
+    /// Executes a method of a WMI class not tied to any specific instance. Examples include
+    /// [Create](https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/create-method-in-class-win32-process) of `Win32_Process`
+    /// and [AddPrinterConnection](https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/addprinterconnection-method-in-class-win32-printer) of `Win32_Printer`.
+    ///
+    /// `MethodClass` should have the name of the class on which the method is being invoked.
+    /// `In` and `Out` can be `()` or any custom structs supporting (de)serialization containing the input and output parameters of the function.
+    ///
+    /// Arrays, Options, unknowns, and nested objects cannot be passed as input parameters due to limitations in how variants are constructed by `windows-rs`.
+    ///
+    /// This function uses [`WMIConnection::exec_instance_method`] internally, with the name of the method class being the instance path, as is expected by WMI.
+    ///
+    /// ```edition2021
+    /// use serde::{Deserialize, Serialize};
+    /// use wmi::{COMLibrary, Variant, WMIConnection, WMIResult};
+    ///
+    /// #[derive(Serialize)]
+    /// #[serde(rename_all = "PascalCase")]
+    /// struct CreateInput {
+    ///     command_line: String
+    /// }
+    ///
+    /// #[derive(Deserialize)]
+    /// #[serde(rename_all = "PascalCase")]
+    /// struct CreateOutput {
+    ///     return_value: u32,
+    ///     process_id: u32
+    /// }
+    ///
+    /// #[derive(Deserialize)]
+    /// #[serde(rename = "Win32_Process")]
+    /// struct Win32Process;
+    ///
+    /// fn main() -> WMIResult<()> {
```

Comment:
Use `/// # ` to hide the non-relevant parts of the code (see `async_query.rs` for more examples).